pkg/migration/controllers/migration.go

```diff
@@ -1605,23 +1638,32 @@ func (m *MigrationController) applyResources(
 		if migration.Spec.IncludeVolumes != nil && !*migration.Spec.IncludeVolumes {
 			pv.Spec.PersistentVolumeReclaimPolicy = v1.PersistentVolumeReclaimRetain
 		}
-		if _, err = adminClient.CoreV1().PersistentVolumes().Update(context.TODO(), &pv, metav1.UpdateOptions{}); err != nil {
-			return err
-		}
+		respPV, err := adminClient.CoreV1().PersistentVolumes().Get(context.TODO(), pv.Name, metav1.GetOptions{})
 		if err != nil {
+			msg := fmt.Errorf("error in reading pv %s during migration: %v", pv.GetName(), err)
 			m.updateResourceStatus(
 				migration,
 				obj,
 				stork_api.MigrationStatusFailed,
-				fmt.Sprintf("Error updating pv resource: %v", err))
-		} else {
+				fmt.Sprintf("Error applying resource: %v", msg))
+			continue
+		}
+		if _, err = adminClient.CoreV1().PersistentVolumes().Update(context.TODO(), respPV, metav1.UpdateOptions{}); err != nil {
```

Comment:
You are not updating the `respPV` object here. The reclaim policy is set on `pv`.